src/raw/bson_ref.rs

```diff
@@ -460,6 +460,14 @@ pub struct RawBinaryRef<'a> {
 }
 
 impl<'a> RawBinaryRef<'a> {
+    /// Copy the contents into a `Binary`.
+    pub fn to_binary(&self) -> Binary {
+        Binary {
+            subtype: self.subtype,
+            bytes: self.bytes.to_owned(),
```

Comment:
WDYT about also implementing the [`ToOwned`](https://doc.rust-lang.org/std/borrow/trait.ToOwned.html) / [`Borrow`](https://doc.rust-lang.org/std/borrow/trait.Borrow.html) relationship for these two types similar to what we do for `RawDocument`/`RawDocumentBuf`?

Reply:
AFAICT that won't work because `Borrow` requires that the signature be `fn borrow(&self) -> &Borrowed`, and because `RawBsonRef` is a wrapper rather than an unsized type that won't work.